worker/uniter/runner/context/context.go

```diff
@@ -982,17 +990,33 @@ func (ctx *HookContext) UpdateSecret(uri *coresecrets.URI, args *jujuc.SecretUpd
 			return ErrIsNotLeader
 		}
 	}
-	ctx.secretChanges.update(uniter.SecretUpdateArg{
+	updateArg := uniter.SecretUpdateArg{
 		SecretUpsertArg: uniter.SecretUpsertArg{
 			URI:          uri,
 			RotatePolicy: args.RotatePolicy,
 			ExpireTime:   args.ExpireTime,
 			Description:  args.Description,
 			Label:        args.Label,
-			Value:        args.Value,
 		},
 		CurrentRevision: md.LatestRevision,
-	})
+	}
+	if args.Value != nil && !args.Value.IsEmpty() {
+		checksum, err := args.Value.Checksum()
+		if err != nil {
+			return errors.Annotate(err, "calculating secret checksum")
+		}
+		if !ok || md.LatestChecksum != checksum {
```

Comment:
Can this `ok` variable be given a better name? It could easily be overridden by a future code change, since it is so commonly used.

Reply:
yup